pkg/recipes/driver/types.go

```diff
@@ -42,6 +42,16 @@ type Driver interface {
 	GetRecipeMetadata(ctx context.Context, opts BaseOptions) (map[string]any, error)
 }
 
+// DriverWithSecrets is an optional interface and used when the driver needs to load secrets for recipe deployment.
+type DriverWithSecrets interface {
+	// Driver is an interface to implement recipe deployment and recipe resources deletion.
+	Driver
+
+	// FindSecretIDs gets the secret ID references associated with git private terraform repository source.
+	// In the future it will be extended to get secret references for provider secrets.
+	FindSecretIDs(ctx context.Context, config recipes.Configuration, definition recipes.EnvironmentDefinition) (string, error)
```

Comment:
Can we generate mocks for this and leverage that for unit testing in engine? I'm not sure why go generate was originally added to the [bicep driver](https://github.com/radius-project/radius/blob/9de3a3e02eec675bc408aa88d57f41ac87baa7b9/pkg/recipes/driver/bicep.go#L48) instead of the types file.

Reply:
added mocks and unit tests